File: week07/5/prob1.cpp
#include <vector>
using namespace std;

class Solution {
public:
    vector<int> getRow(int rowIndex) {
        vector<int> row = vector<int>(rowIndex+1, 1);
        int temp = 1;
        for (size_t i=0; i<=rowIndex; i++) {
            for (size_t j=1; j<i; j++) {
                row.at(j) += temp;
                temp = row.at(j) - temp;
            }
        }
        return row;
    }
};

File: week07/26/prob1.py
class Solution:
    def getRow(self, rowIndex: int) -> List[int]:
        dp = [[] for _ in range(rowIndex+1)]
        dp[0] = [1]
        for i in range(1, rowIndex + 1):
            dp[i] += [1]
            for j in range(1, len(dp[i-1])):
                dp[i] += [dp[i-1][j-1] + dp[i-1][j]]
            dp[i] += [1]
        return dp[-1]